py_hamt/sharded_zarr_store.py

```
@@ -0,0 +1,1011 @@
+            )
+        if not shard_cid:
+            return
+
+        try:
+            shard_data = await self.cas.load(shard_cid)
+            for i in range(0, len(shard_data), self._cid_len):
+                cid_bytes = shard_data[i : i + self._cid_len]
+                if not all(b == 0 for b in cid_bytes):
+                    yield cid_bytes.decode("ascii").rstrip("\x00")
+        except Exception as e:
+            logging.warning(f"Could not process shard {shard_cid} for iteration: {e}")
```

Ruff (0.11.9)

Undefined name `logging`

(F821)